command/command.go

```diff
@@ -229,3 +237,62 @@ func computeNewlineTargets(before, after string) []int {
 
 	return result
 }
+
+// computeMedianIndent determines the median indent of the given string yaml content.
+func computeMedianIndent(yamlContent string) int {
+	lines := strings.Split(yamlContent, "\n")
+	allSpaces := []int{}
+	lastLine := 0
+	for i := 1; i < len(lines); i++ {
+		lineBefore := lines[lastLine]
+		lineAfter := lines[i]
+		// Skip empty lines
+		if strings.TrimSpace(lineBefore) == "" {
+			lastLine++
+			continue
+		}
+		// Do not compare empty lines.
+		if strings.TrimSpace(lineAfter) == "" {
+			continue
+		}
+		l0 := leadingSpaces(lineBefore)
+		l1 := leadingSpaces(lineAfter)
+		lastLine = i
+		// Skip no change in indent.
+		if l0 == l1 {
+			continue
+		}
+		allSpaces = append(allSpaces, absDiffInt(l1, l0))
+	}
+	sort.Sort(sort.IntSlice(allSpaces))
+	if len(allSpaces) == 0 {
+		return defaultIndent
+	}
+	if len(allSpaces)%2 == 0 {
+		// Normally the median for an even number of items would be the average
+		// of the two numbers in the middle. Instead we pick the first of 2
+		// values so that we end up with a reasonable indent.
+		return allSpaces[(len(allSpaces)/2)-1]
+	}
+	return allSpaces[len(allSpaces)/2]
+}
+
+func leadingSpaces(v string) int {
+	cnt := 0
+	for _, c := range v {
+		if c == ' ' {
+			cnt++
+		} else {
+			return cnt
+		}
+	}
+	return cnt
+}
```

Comment:
```suggestion
func leadingSpaces(v string) int {
	cnt := 0
	for _, c := range v {
		if c != ' ' {
			return cnt
		}
		cnt++
	}
	return cnt
}
```

But then how close are you from this?


```suggestion
func leadingSpaces(v string) int {
	for cnt, c := range v {
		if c != ' ' {
			return cnt
		}
	}
	return cnt
}
```

Reply:
Also the function should call strings.TrimSpaceRight

Otherwise a line with 10 spaces wnd only that will report 10